### ISSUE TITLE
[a11y]: ComboBox component not announced selected option with VoiceOver

### ISSUE THREAD

**Issue by patrickobrien1** [carbon-design-system/carbon#19829]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.76.0

### React version

16.14.0

### Automated testing tool and ruleset

VoiceOver

### Assistive technology

VoiceOver

### Description

The ComboBox component does not announce anything indicating that it is selected similar to how the Dropdown component works. I would expect that, when highlighting the currently selected option in the combo box the screen reader will indicate to the user that they are currently over the selected option.

![Image](https://github.com/user-attachments/assets/766dc42c-d52f-4cdc-8bb4-c8bbdba92dbe)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-combobox--default

### Steps to reproduce

1. Instantiate a ComboBox component
2. Select an element within the combo box
3. Activate VoiceOver and highlight the selected element

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
related issue in downshift components https://github.com/carbon-design-system/carbon/issues/19387
